samples/utils/CommandLineUtils.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/crt/Types.h>
+
+namespace Utils
+{
+    /**
+     * A struct to hold the command line options that can be passed to the program from the terminal/console.
+     */
+    struct CommandLineOption
+    {
+        Aws::Crt::String CommandName;
+        Aws::Crt::String ExampleInput;
+        Aws::Crt::String HelpOutput;
+
+        CommandLineOption(Aws::Crt::String p_Name, Aws::Crt::String p_ExampleInput, Aws::Crt::String p_Help = "")
```

Comment:
It doesn't matter at all performance-wise, but from a C++ parameter passing perspective, you're passing strings by value and generating unnecessary copies with a direct assignment.

Pre-C++-11 you'd always pass by const ref in this situation.  I think these days, the current "wisdom/guidance" from the C++ guru camp is pass by value but assign by move:

m_commandName = std::move(p_Name);

ostensibly in the name of supporting parameters that are rvals and certain compiler optimizations when passing small structs.

So while it doesn't matter much in this case, it's good to get in the practice/habit of evaluating the parameter ownership semantics and following the appropriate widely-accepted pattern.